File: Front/myrepapp-vite/src/api/productDocs.ts
// API 基础配置
// 在开发环境使用 localhost，在生产环境使用空字符串（相对路径）
// 使用 ?? 运算符来正确处理空字符串
const API_BASE_URL = import.meta.env.VITE_API_BASE_URL ?? 'http://localhost:8000';

// 产品文档生成请求接口
export interface GenerateProductDocsRequest {
  companyName: string;
  productName: string;
  productCategory: string;
  targetCountry: string;
  productDescription: string;
  language?: string; // 添加语言参数，可选，默认为 'zh'
}

// 合规项目接口
export interface ComplianceItem {
  name: string;
  description: string;
  required: boolean;
  status: string;
}

// 法规详情接口
export interface RegulationDetails {
  countryName: string;
  mainRegulations: string[];
  customsRequirements: string;
  importRestrictions: string;
}

// 产品文档数据接口
export interface ProductDocsData {
  documentContent: string;
  requiredCerts: string[];
  complianceItems: ComplianceItem[];
  estimatedTime: string;
  estimatedCost: string;
  recommendedSteps: string[];
  regulationDetails: RegulationDetails;
}

// 产品文档生成响应接口
export interface GenerateProductDocsResponse {
  code: number;
  message: string;
  data?: ProductDocsData;
}

// 生成产品出口文档
export const generateProductDocs = async (
  request: GenerateProductDocsRequest
): Promise<GenerateProductDocsResponse> => {
  try {
    console.log('发送请求:', request);
    console.log('请求 URL:', `${API_BASE_URL}/product/generate-docs`);
    
    const response = await fetch(`${API_BASE_URL}/product/generate-docs`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(request),
    });

    console.log('响应状态:', response.status);
    console.log('响应头:', response.headers);

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const text = await response.text();
    console.log('响应文本:', text);
    
    // 如果响应为空，返回错误
    if (!text || text.trim() === '' || text.trim() === '{}') {
      throw new Error('服务器返回空响应，请检查后端日志');
    }

    const data = JSON.parse(text);
    console.log('解析后的数据:', data);
    return data;
  } catch (error) {
    console.error('生成产品文档失败:', error);
    throw error;
  }
};

// 导出 API 基础 URL（用于其他可能的 API 调用）
export { API_BASE_URL };

File: GoServer/api/v1/product_docs.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// GenerateProductDocsReq 生成产品出口文档请求
type GenerateProductDocsReq struct {
	g.Meta             `path:"/product/generate-docs" method:"post" summary:"生成产品出口文档"`
	CompanyName        string `json:"companyName" v:"required" dc:"公司名称"`
	ProductName        string `json:"productName" v:"required" dc:"产品名称"`
	ProductCategory    string `json:"productCategory" v:"required" dc:"产品类别"`
	TargetCountry      string `json:"targetCountry" v:"required" dc:"目标国家"`
	ProductDescription string `json:"productDescription" v:"required" dc:"产品描述"`
	Language           string `json:"language" dc:"语言(zh/en)，默认zh"`
}

// GenerateProductDocsRes 生成产品出口文档响应
type GenerateProductDocsRes struct {
	Code    int              `json:"code" dc:"状态码"`
	Message string           `json:"message" dc:"消息"`
	Data    *ProductDocsData `json:"data,omitempty" dc:"文档数据"`
}

// ProductDocsData 产品文档数据
type ProductDocsData struct {
	DocumentContent   string             `json:"documentContent" dc:"文档内容"`
	RequiredCerts     []string           `json:"requiredCerts" dc:"所需认证"`
	ComplianceItems   []ComplianceItem   `json:"complianceItems" dc:"合规项目"`
	EstimatedTime     string             `json:"estimatedTime" dc:"预计时间"`
	EstimatedCost     string             `json:"estimatedCost" dc:"预计费用"`
	RecommendedSteps  []string           `json:"recommendedSteps" dc:"推荐步骤"`
	RegulationDetails *RegulationDetails `json:"regulationDetails" dc:"法规详情"`
}

// ComplianceItem 合规项目
type ComplianceItem struct {
	Name        string `json:"name" dc:"项目名称"`
	Description string `json:"description" dc:"项目描述"`
	Required    bool   `json:"required" dc:"是否必需"`
	Status      string `json:"status" dc:"状态"`
}

// RegulationDetails 法规详情
type RegulationDetails struct {
	CountryName         string   `json:"countryName" dc:"国家名称"`
	MainRegulations     []string `json:"mainRegulations" dc:"主要法规"`
	CustomsRequirements string   `json:"customsRequirements" dc:"海关要求"`
	ImportRestrictions  string   `json:"importRestrictions" dc:"进口限制"`
}
